libraries/scripting/dependencies/src/kotlin/script/experimental/dependencies/annotations.kt

```diff
@@ -33,22 +31,42 @@ annotation class Repository(vararg val repositoriesCoordinates: String)
 /**
  * An extension function that configures repositories and resolves artifacts denoted by the [Repository] and [DependsOn] annotations
  */
-suspend fun ExternalDependenciesResolver.resolveFromAnnotations(annotations: Iterable<Annotation>): ResultWithDiagnostics<List<File>> {
-    annotations.forEach { annotation ->
-        when (annotation) {
-            is Repository -> {
-                for (coordinates in annotation.repositoriesCoordinates) {
-                    if (!tryAddRepository(coordinates))
-                        return makeFailureResult("Unrecognized repository coordinates: $coordinates")
-                }
+suspend fun ExternalDependenciesResolver.resolveFromScriptSourceAnnotations(
+    annotations: Iterable<ScriptSourceAnnotation<*>>
+): ResultWithDiagnostics<List<File>> {
+    return annotations.flatMapSuccess { (annotation, location) ->
+        resolveFromAnnotation(annotation, location)
+    }
+}
+
+/**
+ * An extension function that configures repositories and resolves artifacts denoted by the [Repository] and [DependsOn] annotations
+ */
+suspend fun ExternalDependenciesResolver.resolveFromAnnotations(
+    annotations: Iterable<Annotation>
+): ResultWithDiagnostics<List<File>> {
+    return annotations.flatMapSuccess { resolveFromAnnotation(it, null) }
+}
+
+private suspend fun ExternalDependenciesResolver.resolveFromAnnotation(
```

Comment:
You changed the logic of the annotation processing here. Before we were processing all `@Repository` annotations first and then all `@DependsOn` and now they are processed in one cycle. Which is also ok and maybe even result in more expected behavior, but if you do this, you need to add tests with `@DependsOn`, `@Repository`, `@DependsOn` sequence, to make sure that resolvers can accept repos after a first resolution. We need this test both for the maven and ivy (main-kts) resolvers.
Or you can return to the old logic, but then please add a TODO that proposes this change with a note about tests, as I described it.

Reply:
I actually prefer the previous behavior more. It seems like it just allows for more flexibility, so I'll add a test for ivy and maven that tests that all repositories are added before

Reply:
✅